keras_nlp/samplers/top_k_sampler.py

```diff
@@ -36,56 +33,44 @@ class TopKSampler(Sampler):
     Args:
         k: int, the `k` value of top-k.
         seed: int, defaults to None. The random seed.
-        {{base_sampler_args}}
 
     Call Args:
         {{call_args}}
 
     Examples:
     ```python
-    VOCAB_SIZE = 10
-
-    # Create a dummy model to predict the next token.
-    model = keras.Sequential(
-        [
-            keras.Input(shape=[None]),
-            keras.layers.Embedding(
-                input_dim=VOCAB_SIZE,
-                output_dim=16,
-            ),
-            keras.layers.Dense(VOCAB_SIZE, activation="softmax"),
-        ]
+    # Use a simple alphabet of lowercase characters to [0, 26).
+    int_lookup = {i: chr(i + ord('a')) for i in range(26)}
+    char_lookup = {v: k for k, v in int_lookup.items()}
+    batch_size, length, vocab_size = 1, 12, len(int_lookup)
+
+    def next(prompt, state, index):
+        # A uniform distribution over our alphabet.
+        probs = tf.ones((batch_size, vocab_size))
+        return probs, state
+
+    output = keras_nlp.samplers.TopKSampler(k=3)(
+        next=next,
+        prompt=tf.fill((batch_size, length,), char_lookup['z']),
+        index=5,
     )
-
-    # Define a function that outputs the next token's probability for each token
-    # in the input sequence.
-    def token_probability_fn(inputs, mask):
-        return model(inputs)
-
-    prompt = tf.fill((8, 1), 1)
-
-    sampler = keras_nlp.samplers.TopKSampler(k=5)
-    # Print the generated sequence (token ids).
-    print(sampler(prompt, token_probability_fn, max_length=10))
+    print(["".join([int_lookup[i] for i in s]) for s in output.numpy()])
+    # >>> "zzzzzacbbcaa"
     ```
     """
 
     def __init__(
         self,
         k=5,
         seed=None,
-        jit_compile=True,
-        run_eagerly=False,
     ):
+        super().__init__()
         self.k = k
         self.seed = seed
-        super().__init__(jit_compile=jit_compile, run_eagerly=run_eagerly)
 
-    def get_next_token(self, next_token_probs):
+    def get_next_token(self, probs):
```

Comment:
Generic feedback -- use fully spelled out argument names for anything that isn't a super well established convention. Probabilities / logits here (which one is it?)

Reply:
This is a good comment. I wonder if the thing to do is to remove the `from_logits=True` argument in our samplers. Then things get really obvious. You are either working with `logits` or `probabilities` and never both. Sampling is so complex I think anything we can do to remove cognitive load is worth it.

This would also have the advantage of making it really easy to add a `temperature` argument to all our samplers which scales the logits pre-softmax. (and tightens or loosens the distribution)

Reply:
Added a commit trying this out.